### PR TITLE
Allow duplicate marker type on a portal

### LINKED ISSUE #319: Marker: support duplicate marker type on single portal
This was a limitation in early version of the server that was lifted some versions ago.
This is different from multi-assignments.

**Use cases**
 - Meet up at different deltas at the same portal (or farm/upgrade)
 - Destroy/Capture/Virus/Repeat
 - Multiple Other markers
 - User imagination

### PR DESCRIPTION
**Pull request by bgillock**
Fixes #319 by removing checks for existing marker of same type in type select list and when adding a marker.

### CODE REVIEW

File: src/code/dialogs/markerAddDialog.js [hunk 1]
```diff
@@ -81,8 +81,7 @@ const MarkerAddDialog = WDialog.extend({
         const o = L.DomUtil.create("option", null, this._type);
         o.value = k;
         o.textContent = wX(k);
-        if (markers.has(k)) o.disabled = true;
-        else if (!defaultType) defaultType = k;
+        if (!defaultType) defaultType = k;
```
Comment (le-jeu) [inline src/code/dialogs/markerAddDialog.js:84]: `markers` shouldn't be used in the previous line too
then `defaultType` would always be defined

**Comment by le-jeu** [comment]
Other related parts:

https://github.com/wasabee-project/Wasabee-IITC/blob/bd3020fd92f351f9a6b1da132f2d88fb5459a238/src/code/dialogs/markerChangeDialog.js#L38-L45

This function is bound to disappear
https://github.com/wasabee-project/Wasabee-IITC/blob/bd3020fd92f351f9a6b1da132f2d88fb5459a238/src/code/model/operation.ts#L381-L390

But that needs work in the `applyChanges` method

**Comment by bgillock** [comment]
We can keep getPortalMarkers if we use the m.ID as the key instead of the type. Then make the appropriate changes in applyChanges.

**Comment by le-jeu** [comment]
that won't be really useful for the `applyChanges` function, `getPortalMakers` is there for two reasons: 
 - support old OP (will be removed), since those OP have only one marker type per portal, it's fine to use current code
 - avoid marker edit leading to duplicate markers with same type, this can be removed safely (`summary.edition.duplicate`)

**Comment by le-jeu** [comment]
No, if you change `getPortalMakers` you need to udate every uses. But this function can be keep as it was, since it's valid where it's used in `applyChanges` (1.).
It's just that the mechanism that removes duplicates marker due to marker edit can be removed (2.)

**Pull request merged**